torch/_custom_op/autograd.py

```diff
@@ -226,10 +221,6 @@ def error(what):
             error(f"got object of type {type(grad)} as the gradient for input "
                   f"'{name}', "
                   f"but expected the gradient to be either None or a Tensor")
-        if arg_info != torch.Tensor:
-            error(f"got a Tensor as the gradient for input '{name}' but "
-                  f"expected None as the gradient because input '{name}' "
-                  f"was not a Tensor (it was type {arg_info}).")
```

Comment:
What this code is doing is:
- earlier, we recorded the types of all the args to the operator. the types are listed in arg_info
- during the backward pass, this check is checking that if the gradient is a Tensor, then the arg we are computing the gradient for is also a Tensor

It's not really clear to me what exactly the problem you're running into is, but it is something along the lines of "this check for if the original arg was a Tensor is too restrictive". To fix the problem, we need to relax the check.

To relax the check, we should instead change arg_info to be a boolean if the input is a {Tensor, Tensor subclass} instead of storing the type of the input.